### DISCUSSION TITLE
How to get different response code for "Sent" mail & "Bounce" mail

### DISCUSSION THREAD

**Discussion by PixeloDigital** [postalserver/postal#2250] [Development discussions]
Hello,
I am getting the same response output for "Send" as well as for "Hard fail" email. How can we differentiate between them as for both the output I am getting same response in my python console. 

Here is the response:

reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
send: 'rcpt TO:<example.com>\r\n'
reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
send: 'data\r\n'
reply: b'354 Go ahead\r\n'
reply: retcode (354); Msg: b'Go ahead'
data: (354, b'Go ahead')
send: b'From: @domainname.com\r\nTo: example.com\r\nSubject: Test\r\n\r\nsome text\r\n.\r\n'
reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
data: (250, b'OK')
send: 'mail FROM:<@domainname.com>\r\n'
{}
reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
send: 'rcpt TO:<example.com>\r\n'
reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
send: 'data\r\n'
reply: b'354 Go ahead\r\n'
reply: retcode (354); Msg: b'Go ahead'
data: (354, b'Go ahead')
send: b'From: @domainname.com\r\nTo: example.com\r\nSubject: Test\r\n\r\nsome text\r\n.\r\n'
reply: b'250 OK\r\n'
reply: retcode (250); Msg: b'OK'
data: (250, b'OK')
send: 'mail FROM:<@domainname.com>\r\n'
{}

Please help in the above subjects.
Thank you in advance.

**Comment by willpower232** [comment]
Messages are queued when you complete the request which means that you have to implement webhooks in order to find out if messages are hard failed or sent successfully.

If messages were sent as part of the request then you would experience all kinds of timeout risks.

**Reply by PixeloDigital** [reply]
I am getting <404 Response> when I am trying with webhooks. 
In webhook_url  I am providing my postal smtp server URL. But when I am using webhooks.site url I am getting <Response 200>.
So how can I get <Response 200> for Send Email and <Response 535> for Bounce Email.

**Reply by willpower232** [reply]
> In webhook_url I am providing my postal smtp server URL

You should be sending your webhooks to your own site or code, not Postal?
